src/passes/Precompute.cpp

```diff
@@ -991,6 +958,29 @@ struct Precompute
     return true;
   }
 
+  bool isValidUTF16Literal(const Literal& value) {
```

Comment:
I think the best place for this might be 

`src/support/string.h, cpp`

alongside the existing `isUTF8`.

Reply:
How do I go about the `Literal` representation and `std::string_view`  or `std:string`? I assume we don't want `isValidUtf16Literal` taking a `Literal` in `string.cpp`.

Reply:
Hmm, we have related logic here:

https://github.com/WebAssembly/binaryen/blob/5f767b73d8dc846db532decc64109d3aab63873c/src/wasm/literal.cpp#L660-L677

Perhaps in this PR we can leave it as is and just add a TODO, something like
```cpp
// TODO: move this logic to src/support/string, and refactor to share code with wasm/literal.cpp string printing's conversion from a Literal to a raw string
```

Reply:
sg. Added a TODO.  Thanks.